src/main/java/Figure.java

```diff
@@ -0,0 +1,46 @@
+public abstract class Figure implements FigureInterface {
+    protected double area;
```

Comment:
lets remove area parameter from Figure class as we obtain it dynamically through `getArea` method